solidity/test/TestPhasedEscrow.js

```diff
@@ -426,7 +425,378 @@ describe("PhasedEscrow", () => {
+        from: owner,
+      })
+
+      expect(
+        await batchedPhasedEscrow.isBeneficiaryApproved(beneficiary1.address)
+      ).to.be.false
+      expect(
+        await batchedPhasedEscrow.isBeneficiaryApproved(beneficiary2.address)
+      ).to.be.true
+      expect(
+        await batchedPhasedEscrow.isBeneficiaryApproved(beneficiary3.address)
+      ).to.be.false
+    })
+
+    it("approves multiple beneficiaries ", async () => {
+      await batchedPhasedEscrow.approveBeneficiary(beneficiary1.address, {
+        from: owner,
+      })
+      await batchedPhasedEscrow.approveBeneficiary(beneficiary2.address, {
+        from: owner,
+      })
+
+      expect(
+        await batchedPhasedEscrow.isBeneficiaryApproved(beneficiary1.address)
+      ).to.be.true
+      expect(
+        await batchedPhasedEscrow.isBeneficiaryApproved(beneficiary2.address)
+      ).to.be.true
+      expect(
+        await batchedPhasedEscrow.isBeneficiaryApproved(beneficiary3.address)
+      ).to.be.false
+    })
+  })
+
+  describe("drawee role", async () => {
+    it("is by default assigned to owner", async () => {
+      expect(await batchedPhasedEscrow.drawee()).to.equal(owner)
+    })
+
+    it("can be transferred by owner", async () => {
+      await batchedPhasedEscrow.setDrawee(updatedDrawee, {from: owner})
+      // ok, no revert
+    })
+
+    it("can be transferred by updated owner", async () => {
+      await batchedPhasedEscrow.transferOwnership(updatedOwner, {from: owner})
+
+      await expectRevert(
+        batchedPhasedEscrow.setDrawee(updatedDrawee, {from: owner}),
+        "Ownable: caller is not the owner"
+      )
+      await batchedPhasedEscrow.setDrawee(updatedDrawee, {
+        from: updatedOwner,
+      })
+      // ok, no revert
+    })
+
+    it("can not be transferred by non-owner", async () => {
+      await expectRevert(
+        batchedPhasedEscrow.setDrawee(updatedDrawee, {from: drawee}),
+        "Ownable: caller is not the owner"
+      )
+    })
+
+    it("can be transferred to another account", async () => {
+      let receipt = await batchedPhasedEscrow.setDrawee(drawee, {
+        from: owner,
+      })
+
+      expect(await batchedPhasedEscrow.drawee()).to.equal(drawee)
+      expectEvent(receipt, "DraweeRoleTransferred", {
+        oldDrawee: owner,
+        newDrawee: drawee,
+      })
+
+      receipt = await batchedPhasedEscrow.setDrawee(updatedDrawee, {
+        from: owner,
+      })
+
+      expect(await batchedPhasedEscrow.drawee()).to.equal(updatedDrawee)
+      expectEvent(receipt, "DraweeRoleTransferred", {
+        oldDrawee: drawee,
+        newDrawee: updatedDrawee,
+      })
+    })
+  })
+
+  describe("batchedWithdraw", async () => {
+    let beneficiaries
+    let amounts
+    let escrowBalance
+
+    beforeEach(async () => {
+      beneficiaries = [
+        beneficiary1.address,
+        beneficiary2.address,
+        beneficiary3.address,
+      ]
+
+      amounts = [100, 200, 300]
```

Comment:
Am I correct that we don't need a mechanism for cap a max amount of KEEP that a beneficiary can receive? Ie.  if a first beneficiary will withdraw 90 instead of 100, then a second beneficiary can withdraw 210.

Reply:
From the perspective of `BatchedPhasedEscrow`, we do not. In `withdraw` we do `token.safeTransfer(address(beneficiary), amount)` so we are not approving anything more than that. Individual beneficiary implementations do approve, but they approve a particular recipient address; for example from `__escrowSentTokens` of `StakingPoolRewardsEscrowBeneficiary`: `token.approve(address(rewards), amount)`

Reply:
okay, makes sense.